src/structures/guild.ts

```diff
@@ -343,6 +305,22 @@ export class Guild extends Base {
       }
     })
   }
+
+  /**
+   * Fulfills promise when guild becomes available
+   */
+  async awaitAvailability(): Promise<Guild> {
+    return await new Promise((resolve, reject) => {
+      if(!this.unavailable) return;
+      const listener = (guild: Guild): void => {
+        if (guild.id === this.id) {
+          this.client.removeListener('guildLoaded', listener);
+          resolve(this);
+        }
+      };
+      this.client.on('guildLoaded', listener);
```

Comment:
If the event never occurs, this listener always stays... memory leaks can be there. So, add a timeout to reject and make it configurable through a param in the method.